units/patch_intent.go

```diff
@@ -196,11 +201,19 @@ func (j *patchIntentProcessor) finishPatch(ctx context.Context, patchDoc *patch.
 		return errors.New("patching is diabled for project")
 	}
 
-	// Get and validate patched config and add it to the patch document
-	project, err := validator.GetPatchedProject(ctx, patchDoc, githubOauthToken)
+	// Get and validate patched config
+	projectData, err := validator.GetPatchedProject(ctx, patchDoc, githubOauthToken)
 	if err != nil {
-		return errors.Wrap(err, errInvalidPatchedConfig)
+		return errors.Wrap(err, "can't get patched config")
 	}
+	project, invalid, err := validator.ValidateProjectPatch(projectData, pref.Identifier)
+	if err != nil {
+		if invalid {
+			return errors.Wrap(err, errInvalidPatchedConfig)
```

Comment:
This error string is used to signal the patch is invalid.
Only send it up if it really is invalid.